### DISCUSSION TITLE
C++ vs NodeJS uWebSockets

### DISCUSSION THREAD

**Discussion by KamyarTaher** [uNetworking/uWebSockets#1423] [Q&A] [answered]
Hello, 

I need a backend with the best performance possible in terms of WebSocket connection/speed and mathematical computation.
So I found uWebSockets that seem to be very promising in terms of socket speed, then I come across the following article stating that C++ performs very poorly and Nodejs extremely well in terms of socket speed/connection.
Has anyone more information about this. Is this legit? How a lower-level language like C++ could perform worse than NodeJS?

https://www.researchgate.net/publication/348993267_An_Analysis_of_the_Performance_of_Websockets_in_Various_Programming_Languages_and_Libraries#fullTextFileContent

**Comment by ghost** [comment, marked as answer]
> Is this legit?

No, they obviously messed up. It makes no logical sense that the base library would perform worse than the wrapped up, scriptable variant that's using the base library.
